src/Models/Hooks/useTimeSeriesData.ts

```diff
@@ -0,0 +1,202 @@
+    adapter,
+    connectionString,
+    quickTimeSpan,
+    twins,
+    pollingInterval
+}: IProp): {
+    query: string;
+    deeplink: string;
+    data: Array<ADXTimeSeries>;
+    fetchTimeSeriesData: () => void;
+    isLoading: boolean;
+} => {
+    const [data, setData] = useState<Array<ADXTimeSeries>>(null);
+    const [query, setQuery] = useState('');
+    const [deeplink, setDeeplink] = useState('');
+    const [connectionToQuery, setConnectionToQuery] = useState<IADXConnection>(
+        null
+    );
+
+    const timeSeriesData = useAdapter({
+        adapterMethod: (params: {
+            query: string;
+            connection: IADXConnection;
+        }) => adapter.getTimeSeriesData(params.query, params.connection),
+        refetchDependencies: [
+            adapter,
+            connectionString,
+            twins,
+            pollingInterval
+        ],
+        isAdapterCalledOnMount: false,
+        isLongPolling: pollingInterval ? true : false,
+        pollingIntervalMillis: pollingInterval
+    });
+
+    const onMountDate = useMemo(() => Date.now(), []);
+
+    /**
+     * After getting time series data from adapter, parse it in a structure of
+     * [timestamp, value] and store it as data to expose for consumers of this hook
+     *  */
+    useEffect(() => {
+        if (timeSeriesData?.adapterResult?.result?.data) {
+            logDebugConsole(
+                'debug',
+                `[END]: Fetching time series data finished: `,
+                timeSeriesData?.adapterResult?.result?.data
+            );
+            setData(timeSeriesData?.adapterResult?.result?.data);
+        } else {
+            setData(null);
+        }
+    }, [timeSeriesData?.adapterResult.result]);
+
+    useEffect(() => {
+        if (connectionString) {
+            setConnectionToQuery(
+                generateADXConnectionFromString(connectionString)
+            );
+        } else {
+            setConnectionToQuery(adapter?.getADXConnectionInformation());
+        }
+    }, [connectionString, adapter]);
+
+    const prevQuery = usePrevious(query);
+    useEffect(() => {
+        let newQuery = '';
+        if (connectionToQuery && quickTimeSpan && twins?.length) {
+            const timeFrom = new Date(onMountDate - quickTimeSpan);
+            const timeTo = new Date(onMountDate);
+            newQuery = getBulkADXQueryFromTimeSeriesTwins(
+                twins,
+                { from: timeFrom, to: timeTo },
+                connectionToQuery
+            );
+        }
+        if (prevQuery !== newQuery) {
+            setQuery(newQuery);
+        }
+    }, [connectionToQuery, quickTimeSpan, twins, onMountDate]);
+
+    const prevDeeplink = usePrevious(deeplink);
+    useEffect(() => {
+        const newDeeplink = getADXDeeplinkForWeb(connectionToQuery, query);
+        if (prevDeeplink !== newDeeplink) {
+            setDeeplink(newDeeplink);
+        }
+    }, [connectionToQuery, query]);
```

Comment:
internally usePrevious keeps the prevvalue as a ref. So no need to add if to dependency list since outside it is kept updated and inside useeffect we get that ref value